### PR TITLE
change: make v2 migration script add version args when needed

### PR DESCRIPTION
**Pull request by metrizable**
*Issue #, if available:* #1465 

*Description of changes:*

changes include:

* transmogrify framework and model constructors to include `framework_version` and `py_version`, as appropriate
* testing

*Testing done:*

unit:

```
% tox --parallel 3 tests/unit
✔ OK black-format in 7.752 seconds
✔ OK flake8 in 21.401 seconds
✔ OK twine in 14.38 seconds
✔ OK pylint in 34.061 seconds
✔ OK doc8 in 19.642 seconds
✔ OK py36 in 52.532 seconds
✔ OK sphinx in 1 minute, 36.107 seconds
✔ OK py37 in 50.303 seconds
✔ OK py38 in 45.694 seconds
✔ OK py27 in 2 minutes, 10.535 seconds
```

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have passed the region in to any/all clients that I've initialized as part of this change.
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker/cli/compatibility/v2/modifiers/framework_version.py [hunk 1]
```diff
@@ -112,30 +80,144 @@ def modify_node(self, node):
...
-        elif isinstance(node.func, ast.Attribute):
-            framework = node.func.attr
 
-        if framework.endswith("Model"):
-            framework = framework[: framework.find("Model")]
+def _py_version_defaults(framework, framework_version, is_model=False):
+    """Gets the py_version required for the framework_version and if it's a model
+
+    Args:
+        framework (str): name of the framework
```
Comment (laurenyu) [inline src/sagemaker/cli/compatibility/v2/modifiers/framework_version.py:109]: this is missing the two other args

File: src/sagemaker/cli/compatibility/v2/modifiers/framework_version.py [hunk 2]
```diff
@@ -112,30 +80,144 @@ def modify_node(self, node):
...
+
+    Returns:
+        str: the default py version, as appropriate. None if no default py_version
+    """
+    if framework in ["Chainer", "PyTorch"]:
```
Comment (laurenyu) [inline src/sagemaker/cli/compatibility/v2/modifiers/framework_version.py:114]: I've been told tuples are more efficient than lists for immutable iterables (though I realize this one's pretty small no matter what)

File: src/sagemaker/cli/compatibility/v2/modifiers/framework_version.py [hunk 3]
```diff
@@ -112,30 +80,144 @@ def modify_node(self, node):
...
+
+def _tf_py_version_default(framework_version):
+    """Gets the py_version default based on framework_version for TensorFlow."""
+    if not framework_version:
+        return "py2"
+    version = [int(s) for s in framework_version.split(".")]
+    if version < [1, 12]:
+        return "py2"
+    if version < [2, 2]:
+        return "py3"
```
Comment (laurenyu) [inline src/sagemaker/cli/compatibility/v2/modifiers/framework_version.py:133]: (optional) you could also use `packaging.version.Version` if you're getting tired of the split/list comprehension

File: tests/unit/sagemaker/cli/compatibility/v2/modifiers/test_framework_version.py [hunk 4]
```diff
@@ -21,135 +21,215 @@
...
+
+TEMPLATES = [
+    Template(
+        framework="TensorFlow",
+        framework_version="1.11.0",
+        py_version="py2",
+        framework_templates=[
+            "TensorFlow({})",
+            "sagemaker.tensorflow.TensorFlow({})",
+            "sagemaker.tensorflow.estimator.TensorFlow({})",
```
Comment (laurenyu) [inline tests/unit/sagemaker/cli/compatibility/v2/modifiers/test_framework_version.py:87]: (out of scope) have you thought about generating the variations as well? the only thing different from framework to framework for these is the framework name

File: tests/unit/sagemaker/cli/compatibility/v2/modifiers/test_framework_version.py [hunk 5]
```diff
@@ -21,135 +21,215 @@
...
+            "SKLearnModel({})",
+            "sagemaker.sklearn.SKLearnModel({})",
+            "sagemaker.sklearn.model.SKLearnModel({})",
+        ],
+        py_version_for_model=False,
+    ),
+]
+
+
+def get_constructors(versions=False, image=False):
```
Comment (laurenyu) [inline tests/unit/sagemaker/cli/compatibility/v2/modifiers/test_framework_version.py:160]: I believe it's less Pythonic for a function to start with "get" - it's considered redundant?

File: src/sagemaker/cli/compatibility/v2/modifiers/framework_version.py (hunk 1 above)
Comment (metrizable) [reply src/sagemaker/cli/compatibility/v2/modifiers/framework_version.py:109]: will update

File: src/sagemaker/cli/compatibility/v2/modifiers/framework_version.py (hunk 2 above)
Comment (metrizable) [reply src/sagemaker/cli/compatibility/v2/modifiers/framework_version.py:114]: agreed.

File: src/sagemaker/cli/compatibility/v2/modifiers/framework_version.py (hunk 3 above)
Comment (metrizable) [reply src/sagemaker/cli/compatibility/v2/modifiers/framework_version.py:133]: will update

File: tests/unit/sagemaker/cli/compatibility/v2/modifiers/test_framework_version.py (hunk 4 above)
Comment (metrizable) [reply tests/unit/sagemaker/cli/compatibility/v2/modifiers/test_framework_version.py:87]: good call

File: tests/unit/sagemaker/cli/compatibility/v2/modifiers/test_framework_version.py (hunk 5 above)
Comment (metrizable) [reply tests/unit/sagemaker/cli/compatibility/v2/modifiers/test_framework_version.py:160]: removing

File: src/sagemaker/cli/compatibility/v2/modifiers/framework_version.py (hunk 2 above)
Comment (metrizable) [reply src/sagemaker/cli/compatibility/v2/modifiers/framework_version.py:114]: and arguably for `in` checks, `set`

**Pull request merged**